### PR TITLE
Fixed fmtNumber returning '-0' when rounding small negative numbers

### PR DESCRIPTION
**Pull request by jskupsik**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by jskupsik** [comment]
Before: 
```js
fmtNumber(-0.0004, {precision: 2}) === '-0.00'
```

After:
```js
fmtNumber(-0.0004, {precision: 2}) === '0.00'
```

**Review by Ryanseanlee — APPROVED** [review]

**Comment by lbwexler** [comment]
The implementation of this seems good, although I might be inclined to keep the initial early test for a true zero, for clarity/efficiency.

My bigger question is just to be sure that this is what we want.  I *think* its correct and appropriate, but to be clear we are throwing away a bit of information as to the fact that it is a diminuous amount with a real sign

**Comment by lbwexler** [comment]
@jskupsik -- we discussed this and came up with the following:

1) Would like this behavior to be goverened by a new flag.  Not sure about name.  `displayRoundToZeroAsZero`?  `strictZeroDisplay`?  `zeroDisplayMode`.
2) In looking at the implementation, it seems like the logic will not prevent a ledger formatted number of the form `(0.00)`, if there is no zero display.  That seems inconsistent. Can we fix?

**Comment by jskupsik** [comment]
fmtNumber(-0.0004, {precision:2})
'-0.00'
fmtNumber(-0.0004, {precision:2, strictZeroUnsigned: true})
'0.00'

fmtNumber(0.0004, {precision:2, withPlusSign: true})
'+0.00'
fmtNumber(0.0004, {precision:2, withPlusSign: true, strictZeroUnsigned: true})
'0.00'

fmtNumber(-0.0004, {precision:2, ledger: true})
-> children: (3) ['(', '0.00', ')']

fmtNumber(-0.0004, {precision:2, ledger: true, strictZeroUnsigned: true})
-> children: (2) ['0.00', {…}]

File: format/FormatNumber.ts [hunk 1]
```diff
@@ -53,6 +53,9 @@ export interface NumberFormatOptions extends Omit<FormatOptions<number>, 'toolti
     /** True to prepend positive numbers with a '+'. */
     withPlusSign?: boolean;
 
+    /** True to not display a signed zero -0.0 (and +0.0 withPlusSign) in cases where a small number is rounded to zero. */
+    strictZeroUnsigned?: boolean;
+
```
Comment (lbwexler) [inline format/FormatNumber.ts:58]: how about something like `strictZero`?  Captures the fact that this relates to interpretation of numbers rounding to zero generally, both for the purposes of zeroDisplay and the signing.   But I think the name should imply the reverse sense.  That is 'strict' means that only true zero gets treated like zero

Comment (lbwexler) [reply format/FormatNumber.ts:58]: other than that, implementation looks good!

**Comment by jskupsik** [comment]
Ran the same 6 tests as above.

**Pull request merged**